lib/view_component/action_view_compatibility.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module ViewComponent
+  module ActionViewCompatibility
+    extend ActiveSupport::Concern
+
+    def form_for(*args, **kwags, &block)
+      with_compatible_capture do
+        super
+      end
+    end
+
+    def form_with(*args, **kwargs, &block)
+      with_compatible_capture do
+        super
+      end
+    end
+
+    included do
+      alias_method :original_capture, :capture
+      alias_method :capture, :capture_with_compatibility
+    end
+
+    def with_compatible_capture(&block)
+      old_compatbile_capture = defined?(@compatbile_capture) ? @compatbile_capture : false
+      @compatbile_capture = true
+      yield
+    ensure
+      @compatbile_capture = old_compatbile_capture
```

Comment:
```suggestion
      old_compatible_capture = defined?(@compatible_capture) ? @compatible_capture : false
      @compatible_capture = true
      yield
    ensure
      @compatible_capture = old_compatible_capture
```

Reply:
Thanks! I typo'd it once and my editor decided that was the way to autocomplete it from then on. 😅